File: python/659_分割数组为连续子序列_medium.py
class Solution:
    def isPossible(self, nums: List[int]) -> bool:
        cnt = collections.Counter(nums)
        ent = collections.Counter()
        for e in nums:
            if not cnt[e]:
                continue
            elif ent[e-1]:
                cnt[e], ent[e-1], ent[e] = cnt[e]-1, ent[e-1]-1, ent[e]+1
            elif cnt[e+1] and cnt[e+2]:
                cnt[e], cnt[e+1], cnt[e+2] = cnt[e]-1, cnt[e+1]-1, cnt[e+2]-1
                ent[e+2] += 1
            else: return False
        return True

File: cpp/659_分割数组为连续子序列_medium.cpp
class Solution {
public:
    bool isPossible(vector<int>& nums) {
        unordered_map <int, int> cnt, ent;
        for(int el : nums)
            cnt[el]++;
        for(int el : nums)
            if(!cnt[el])
                continue;
            else if(ent[el-1]){
                ent[el-1]--;
                ent[el]++;
                cnt[el]--;
            }
            else if(cnt[el+1] && cnt[el+2]){
                cnt[el]--;
                cnt[el+1]--;
                cnt[el+2]--;
                ent[el+2]++;
            }
            else return false;
        return true;
    }
};